subsys/net/lib/nrf_cloud/coap/src/nrf_cloud_coap_transport.c

```diff
@@ -642,10 +643,57 @@ int nrf_cloud_coap_transport_init(struct nrf_cloud_coap_client *const client)
 	return err;
 }
 
+static int nrf_cloud_coap_try_address(struct sockaddr *const addr)
+{
+	int err;
+	int sock;
+	size_t addr_size;
+
+	LOG_DBG("Creating socket type IPPROTO_DTLS_1_2");
+	sock = socket(addr->sa_family, SOCK_DGRAM, IPPROTO_DTLS_1_2);
+
+	if (sock < 0) {
+		LOG_INF("Failed to create CoAP socket, errno: %d", errno);
+		err = -ENOTSOCK;
+		goto out;
+	}
+
+	LOG_DBG("sock = %d", sock);
+
+	err = nrfc_dtls_setup(sock);
+	if (err < 0) {
+		LOG_INF("Failed to initialize the DTLS client: %d", err);
+		err = -EPROTO;
+		goto out;
+	}
+
+	addr_size = sizeof(struct sockaddr_in6);
+	if (addr->sa_family == AF_INET) {
+		addr_size = sizeof(struct sockaddr_in);
+	}
+
+	err = connect(sock, addr, addr_size);
+
+	if (err) {
+		LOG_INF("Connect failed, errno: %d", errno);
+		err = -ECONNREFUSED;
+		goto out;
+	}
+
+out:
+	if (err && sock >= 0) {
```

Comment:
if socket creation fails, `sock` is returned

Reply:
Ah, yes, I need to split this if statement, thanks.